jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -2268,6 +2269,123 @@ ecma_builtin_array_prototype_object_find (ecma_value_t predicate, /**< callback
   /* 9. */
   return is_find ? ECMA_VALUE_UNDEFINED : ecma_make_integer_value (-1);
 } /* ecma_builtin_array_prototype_object_find */
+
+/**
+ * The Array.prototype object's 'copyWithin' routine
+ *
+ * See also:
+ *          ECMA-262 v6, 22.1.3.3
+ *
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_array_prototype_object_copy_within (const ecma_value_t args[], /**< arguments list */
+                                                 ecma_length_t args_number, /**< number of arguments */
+                                                 ecma_object_t *obj_p, /**< array object */
+                                                 uint32_t len) /**< array object's length */
+{
+  if (len == 0)
+  {
+    return ecma_make_integer_value (-1);
+  }
+
+  if (args_number == 0)
+  {
+    return ecma_copy_value (ecma_make_object_value (obj_p));
+  }
+
+  ecma_number_t d_len = (double) len;
+  ecma_number_t target;
+  ecma_value_t error = ecma_op_to_integer (args[0], &target);
+
+  if (ECMA_IS_VALUE_ERROR (error))
+  {
+    return error;
+  }
+
+  target = ecma_number_is_negative (target) ? JERRY_MAX (d_len + target, 0) : JERRY_MIN (target, d_len);
+
+  ecma_number_t start = 0;
+  ecma_number_t end = d_len;
```

Comment:
I would prefer if we used `uint32_t`-s, the values are already integers in the [0, UINT32_MAX] range.